aea/cli/generate.py

```diff
@@ -136,6 +145,24 @@ def _generate_item(ctx: Context, item_type, specification_path):
         )
         sys.exit(1)
 
+    # Run black code formatting
+    try:
+        subp = subprocess.Popen(  # nosec
+            [
+                sys.executable,
+                "-m",
+                "black",
+                os.path.join(item_type_plural, protocol_spec.name),
+                "--quiet",
+            ]
+        )
+        subp.wait(5.0)
+    finally:
+        poll = subp.poll()
+        if poll is None:  # pragma: no cover
+            subp.terminate()
+            subp.wait(2)
```

Comment:
in theory, this could fail again if SIGTERM is not enough.

Reply:
but for `black` it should be OK

Reply:
hm ok..we have this pattern all over the place.